decred/decred/dcr/dcrdata.py

```diff
@@ -757,10 +757,9 @@ def updateTip(self):
         """
         try:
             self.tip = self.bestBlock()
-            return
         except Exception as e:
             log.error("failed to retrieve tip from blockchain: %s" % formatTraceback(e))
-        raise DecredError("no tip data retrieved")
+            raise DecredError("no tip data retrieved")
```

Comment:
wonder why I never saw this exception happen?

Reply:
Probably because `updateTip` is called in `connect`, right after `DcrdataClient` creation, and if there's a problem connecting to the service, the latter is going to fail first.